src/erlcloud_ecs.erl

```diff
@@ -0,0 +1,2467 @@
+        {cluster, <<"cluster">>, fun to_binary/1}
+    ].
+
+-spec describe_container_instances_record() -> record_desc().
+describe_container_instances_record() ->
+    {#ecs_describe_container_instances{},
+     [{<<"containerInstances">>, #ecs_describe_container_instances.container_instances, fun decode_container_instances_list/2},
+      {<<"failures">>, #ecs_describe_container_instances.failures, fun decode_failures_list/2}
+     ]}.
+
+-spec describe_container_instances(Instances :: [string_param()]) -> ecs_return(#ecs_describe_container_instances{}).
+describe_container_instances(Instances) ->
+    describe_container_instances(Instances, [], default_config()).
+
+-spec describe_container_instances(
+        Instances :: [string_param()],
+        describe_container_instances_opts() | aws_config()) -> ecs_return(#ecs_describe_container_instances{}).
+describe_container_instances(Instances, #aws_config{} = Config) ->
+    describe_container_instances(Instances, [], Config);
+describe_container_instances(Instances, Opts) ->
+    describe_container_instances(Instances, Opts, default_config()).
+
+%%%------------------------------------------------------------------------------
+%% @doc 
+%% ECS API
+%% [http://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_DescribeContainerInstances.html]
+%%
+%% ===Example===
+%%
+%% Describe ECS Container Instances with the the name "709b350e-8c8e-490c-8fa3-d5fc09ac9e0b" in a "TestCluster" cluster
+%%
+%% `
+%% {ok, Instances} = erlcloud_ecs:describe_container_instances(["709b350e-8c8e-490c-8fa3-d5fc09ac9e0b"], [{cluster, "TestCluster"}, {out, json}]),
+%% '
+%% @end
+%%%------------------------------------------------------------------------------
+-spec describe_container_instances(
+        Instances :: [string_param()],
+        Opts :: describe_container_instances_opts(),
+        Config :: aws_config()) -> ecs_return(#ecs_describe_container_instances{}).
+describe_container_instances(Instances, Opts, #aws_config{} = Config) ->
+    {AwsOpts, EcsOpts} = opts(describe_container_instances_opts(), Opts),
+    Return = ecs_request(
+                Config,
+                "DescribeContainerInstances",
+                [{<<"containerInstances">>, to_binary(Instances)}] ++ AwsOpts),
+    out(Return, fun(Json, UOpts) -> decode_record(describe_container_instances_record(), Json, UOpts) end, 
+        EcsOpts).
+
+%%%------------------------------------------------------------------------------
+%% DescribeServices
+%%%------------------------------------------------------------------------------
+-type describe_services_opt() :: {cluster, string_param()}.
+-type describe_services_opts() :: [describe_services_opt()].
+
+-spec describe_services_opts() -> opt_table().
+describe_services_opts() ->
+    [
+        {cluster, <<"cluster">>, fun to_binary/1}
+    ].
+
+-spec describe_services_record() -> record_desc().
+describe_services_record() ->
+    {#ecs_describe_services{},
+     [{<<"services">>, #ecs_
```

Comment:
comment does not make sense to me. please clarify where are the booleans? http://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_Service.html
http://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_DescribeServices.html